src/EditorFeatures/Core/Implementation/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -39,31 +46,38 @@ protected AbstractGoToDefinitionService(IEnumerable<Lazy<INavigableItemsPresente
         {
             var symbol = await FindSymbolAsync(document, position, cancellationToken).ConfigureAwait(false);
 
+            // Try to compute source definitions from symbol.
+            var items = symbol != null ? NavigableItemFactory.GetItemsFromPreferredSourceLocations(document.Project.Solution, symbol) : null;
+            if (items == null || items.IsEmpty())
+            {
+                // Fallback to asking the navigation definition providers for navigable definition locations.
+                items = await GoToDefinitionHelpers.FindExternalDefinitionsAsync(document, position, _externalDefinitionProviders, cancellationToken).ConfigureAwait(false);
+            }
+
             // realize the list here so that the consumer await'ing the result doesn't lazily cause
             // them to be created on an inappropriate thread.
-            return NavigableItemFactory.GetItemsFromPreferredSourceLocations(document.Project.Solution, symbol).ToList();
+            return items?.ToList();
         }
 
         public bool TryGoToDefinition(Document document, int position, CancellationToken cancellationToken)
         {
+            // First try to compute the referenced symbol and attempt to go to definition for the symbol.
             var symbol = FindSymbolAsync(document, position, cancellationToken).WaitAndGetResult(cancellationToken);
-
             if (symbol != null)
             {
                 var isThirdPartyNavigationAllowed = IsThirdPartyNavigationAllowed(symbol, position, document, cancellationToken);
 
-                if (GoToDefinitionHelpers.TryGoToDefinition(symbol,
+                return GoToDefinitionHelpers.TryGoToDefinition(symbol,
                     document.Project,
+                    _externalDefinitionProviders,
                     _presenters,
                     thirdPartyNavigationAllowed: isThirdPartyNavigationAllowed,
                     throwOnHiddenDefinition: true,
-                    cancellationToken: cancellationToken))
-                {
-                    return true;
-                }
+                    cancellationToken: cancellationToken);
             }
 
-            return false;
+            // Otherwise, fallback to the external navigation definition providers.
```

Comment:
ordering seems sometime external provider go first and sometime builtin goes first. what is the ordering policy?